pkg/bff/users.go

```diff
@@ -54,121 +68,213 @@ func (s *Server) Login(c *gin.Context) {
 		return
 	}
 
-	// Retrieve the user's organization from the database
 	var org *models.Organization
-	if appdata.OrgID == "" {
-		// Create the organization
+	if params.OrgID == "" && appdata.OrgID == "" {
+		// This is a new user so create a new organization for them
 		org, err = s.db.CreateOrganization()
 		if err != nil {
 			log.Error().Err(err).Msg("could not create organization for new user")
-			c.JSON(http.StatusInternalServerError, "could not complete user login")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not complete user login"))
 			return
 		}
 
-		// Set the organization ID in the user app metadata
-		appdata.OrgID = org.Id
-	} else {
-		// Organizations are stored by UUID in the database
-		var id uuid.UUID
-		if id, err = models.ParseOrgID(appdata.OrgID); err != nil {
-			log.Error().Err(err).Msg("could not parse organization ID")
-			c.JSON(http.StatusInternalServerError, "could not complete user login")
+		// Add the user to the organization in the database
+		collaborator := &models.Collaborator{
+			Email:    *user.Email,
+			UserId:   *user.ID,
+			Verified: *user.EmailVerified,
+		}
+		if err = org.AddCollaborator(collaborator); err != nil {
+			log.Error().Err(err).Str("user_id", collaborator.UserId).Msg("could not add collaborator to organization")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not complete user login"))
 			return
 		}
+	} else {
+		var orgID string
+		if params.OrgID != "" {
+			// Try to login the user to the requested organization if provided
+			orgID = params.OrgID
+		} else {
+			// Default to the last used organization
+			orgID = appdata.OrgID
+		}
 
-		// Get the organization for the specified user
-		org, err = s.db.RetrieveOrganization(id)
-		if err != nil {
-			log.Error().Err(err).Str("orgid", appdata.OrgID).Msg("could not retrieve organization for user VASP verification")
-			c.JSON(http.StatusInternalServerError, "could not complete user login")
+		// Fetch the organization from the database
+		if org, err = s.OrganizationFromID(orgID); err != nil {
+			log.Error().Err(err).Str("org_id", orgID).Msg("could not fetch organization for invited user")
+			c.JSON(http.StatusNotFound, api.ErrorResponse("organization not found"))
 			return
 		}
 
-		// Ensure the VASP record is correct for the user
-		if org.Testnet != nil && org.Testnet.Id != "" {
-			appdata.VASPs.TestNet = org.Testnet.Id
-		}
-		if org.Mainnet != nil && org.Mainnet.Id != "" {
-			appdata.VASPs.MainNet = org.Mainnet.Id
+		// Retrieve the user's collaborator record from the organization
+		var collaborator *models.Collaborator
+		if collaborator = org.GetCollaborator(*user.Email); collaborator == nil {
+			log.Debug().Str("email", *user.Email).Str("org_id", org.Id).Msg("could not find user in organization")
+			c.JSON(http.StatusUnauthorized, api.ErrorResponse("user is not authorized to access this organization"))
+			return
 		}
```

Comment:
This section is critical for security; can we add some comments to it to ensure future maintainers guard against this case; also comments around the test for this case?